pkgs/applications/emulators/libretro/cores/fbalpha2012.nix

```diff
@@ -1,11 +1,14 @@
 {
   lib,
+  gcc13Stdenv,
   fetchFromGitHub,
   mkLibretroCore,
 }:
 mkLibretroCore {
   core = "fbalpha2012";
   version = "0-unstable-2024-10-21";
+  # https://github.com/libretro/fbalpha2012/issues/116
```

Comment:
this failure is a real bug. it assigns an `int*` to a `long*` (which is a crc32 table) then does uses the table to do encrypt / decryption using the wrong crc32 values.  the code vendored from zlib so it is easy enough to replace.

<details><summary>unvendor zlib and minizip</summary>

```diff
diff --git a/pkgs/applications/emulators/libretro/cores/fbalpha2012.nix b/pkgs/applications/emulators/libretro/cores/fbalpha2012.nix
index febf6f7935a8..aadf09cb7822 100644
--- a/pkgs/applications/emulators/libretro/cores/fbalpha2012.nix
+++ b/pkgs/applications/emulators/libretro/cores/fbalpha2012.nix
@@ -2,8 +2,10 @@
   lib,
   fetchFromGitHub,
   mkLibretroCore,
+  zlib,
+  runCommand
 }:
-mkLibretroCore {
+mkLibretroCore rec {
   core = "fbalpha2012";
   version = "0-unstable-2024-10-21";
 
@@ -14,8 +16,27 @@ mkLibretroCore {
     hash = "sha256-giEV09dT/e82bmDlRkxpkW04JcsEZc/enIPecqYtg3c=";
   };
 
+  sourceRoot = "${src.name}/svn-current/trunk";
+
+  # unvendor zlib and broken minizip code
+  postPatch = let
+    minizip-src = runCommand "minizip-src" {} ''
+      mkdir $out
+      unpackFile ${zlib.src}
+      cp */contrib/minizip/{unzip.*,ioapi.*,crypt.h} $out/
+    '';
+  in ''
+    substituteInPlace ${makefile} \
+      --replace-fail '-I$(FBA_LIB_DIR)/zlib' ""
+
+    cp ${minizip-src}/* src/burner
+  '';
+
+  buildInputs = [ zlib ];
+
+  makeFlags = [ "EXTERNAL_ZLIB=1" ];
+
   makefile = "makefile.libretro";
-  preBuild = "cd svn-current/trunk";
 
   meta = {
     description = "Port of Final Burn Alpha ~2012 to libretro";

```
</details>